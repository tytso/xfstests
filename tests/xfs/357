#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2017 Oracle, Inc.  All Rights Reserved.
#
# FS QA Test No. 357
#
# Populate a XFS filesystem and fuzz every AGI field.
# Use xfs_scrub to fix the corruption.
#
. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_scrub dangerous_online_repair

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_fuzz_fields

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Fuzz AGI"
_scratch_xfs_fuzz_metadata '' 'online' 'agi 1' >> $seqres.full
echo "Done fuzzing AGI"

# success, all done
status=0
exit
