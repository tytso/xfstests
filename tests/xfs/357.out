QA output created by 357
Format and populate
Fuzz AGI
magicnum = zeroes: mount failed (32).
magicnum = ones: mount failed (32).
magicnum = firstbit: mount failed (32).
magicnum = middlebit: mount failed (32).
magicnum = lastbit: mount failed (32).
magicnum = add: mount failed (32).
magicnum = sub: mount failed (32).
versionnum = zeroes: mount failed (32).
versionnum = ones: mount failed (32).
versionnum = firstbit: mount failed (32).
versionnum = middlebit: mount failed (32).
versionnum = lastbit: mount failed (32).
versionnum = add: mount failed (32).
versionnum = sub: mount failed (32).
seqno = zeroes: mount failed (32).
seqno = ones: mount failed (32).
seqno = firstbit: mount failed (32).
seqno = middlebit: mount failed (32).
seqno = lastbit: mount failed (32).
seqno = add: mount failed (32).
seqno = sub: mount failed (32).
level = zeroes: mount failed (32).
level = ones: mount failed (32).
level = firstbit: mount failed (32).
level = middlebit: mount failed (32).
level = add: mount failed (32).
level = sub: mount failed (32).
newino = ones: online scrub didn't fail.
newino = middlebit: online scrub didn't fail.
newino = lastbit: online scrub didn't fail.
newino = add: online scrub didn't fail.
newino = sub: online scrub didn't fail.
dirino = add: online scrub didn't fail.
unlinked[0] = zeroes: mount failed (32).
unlinked[0] = firstbit: mount failed (32).
unlinked[0] = middlebit: mount failed (32).
unlinked[0] = lastbit: mount failed (32).
unlinked[0] = add: online scrub didn't fail.
unlinked[0] = add: offline re-scrub failed (1).
unlinked[0] = add: offline post-mod scrub failed (1).
unlinked[0] = sub: mount failed (32).
unlinked[1] = zeroes: mount failed (32).
unlinked[1] = firstbit: mount failed (32).
unlinked[1] = middlebit: mount failed (32).
unlinked[1] = lastbit: mount failed (32).
unlinked[1] = add: online scrub didn't fail.
unlinked[1] = add: offline re-scrub failed (1).
unlinked[1] = add: offline post-mod scrub failed (1).
unlinked[1] = sub: mount failed (32).
unlinked[2] = zeroes: mount failed (32).
unlinked[2] = firstbit: mount failed (32).
unlinked[2] = middlebit: mount failed (32).
unlinked[2] = lastbit: mount failed (32).
unlinked[2] = add: online scrub didn't fail.
unlinked[2] = add: offline re-scrub failed (1).
unlinked[2] = add: offline post-mod scrub failed (1).
unlinked[2] = sub: mount failed (32).
unlinked[3] = zeroes: mount failed (32).
unlinked[3] = firstbit: mount failed (32).
unlinked[3] = middlebit: mount failed (32).
unlinked[3] = lastbit: mount failed (32).
unlinked[3] = add: online scrub didn't fail.
unlinked[3] = add: offline re-scrub failed (1).
unlinked[3] = add: offline post-mod scrub failed (1).
unlinked[3] = sub: mount failed (32).
unlinked[4] = zeroes: mount failed (32).
unlinked[4] = firstbit: mount failed (32).
unlinked[4] = middlebit: mount failed (32).
unlinked[4] = lastbit: mount failed (32).
unlinked[4] = add: online scrub didn't fail.
unlinked[4] = add: offline re-scrub failed (1).
unlinked[4] = add: offline post-mod scrub failed (1).
unlinked[4] = sub: mount failed (32).
unlinked[5] = zeroes: mount failed (32).
unlinked[5] = firstbit: mount failed (32).
unlinked[5] = middlebit: mount failed (32).
unlinked[5] = lastbit: mount failed (32).
unlinked[5] = add: online scrub didn't fail.
unlinked[5] = add: offline re-scrub failed (1).
unlinked[5] = add: offline post-mod scrub failed (1).
unlinked[5] = sub: mount failed (32).
unlinked[6] = zeroes: mount failed (32).
unlinked[6] = firstbit: mount failed (32).
unlinked[6] = middlebit: mount failed (32).
unlinked[6] = lastbit: mount failed (32).
unlinked[6] = add: online scrub didn't fail.
unlinked[6] = add: offline re-scrub failed (1).
unlinked[6] = add: offline post-mod scrub failed (1).
unlinked[6] = sub: mount failed (32).
unlinked[7] = zeroes: mount failed (32).
unlinked[7] = firstbit: mount failed (32).
unlinked[7] = middlebit: mount failed (32).
unlinked[7] = lastbit: mount failed (32).
unlinked[7] = add: online scrub didn't fail.
unlinked[7] = add: offline re-scrub failed (1).
unlinked[7] = add: offline post-mod scrub failed (1).
unlinked[7] = sub: mount failed (32).
unlinked[8] = zeroes: mount failed (32).
unlinked[8] = firstbit: mount failed (32).
unlinked[8] = middlebit: mount failed (32).
unlinked[8] = lastbit: mount failed (32).
unlinked[8] = add: online scrub didn't fail.
unlinked[8] = add: offline re-scrub failed (1).
unlinked[8] = add: offline post-mod scrub failed (1).
unlinked[8] = sub: mount failed (32).
unlinked[9] = zeroes: mount failed (32).
unlinked[9] = firstbit: mount failed (32).
unlinked[9] = middlebit: mount failed (32).
unlinked[9] = lastbit: mount failed (32).
unlinked[9] = add: online scrub didn't fail.
unlinked[9] = add: offline re-scrub failed (1).
unlinked[9] = add: offline post-mod scrub failed (1).
unlinked[9] = sub: mount failed (32).
uuid = zeroes: mount failed (32).
uuid = ones: mount failed (32).
uuid = firstbit: mount failed (32).
uuid = middlebit: mount failed (32).
uuid = lastbit: mount failed (32).
crc = zeroes: mount failed (32).
crc = ones: mount failed (32).
crc = firstbit: mount failed (32).
crc = middlebit: mount failed (32).
crc = lastbit: mount failed (32).
crc = add: mount failed (32).
crc = sub: mount failed (32).
free_level = zeroes: mount failed (32).
free_level = ones: mount failed (32).
free_level = firstbit: mount failed (32).
free_level = middlebit: mount failed (32).
free_level = add: mount failed (32).
free_level = sub: mount failed (32).
Done fuzzing AGI
